Mobile/ShopLens/Droid/Activities/MainActivity.cs

```diff
@@ -401,6 +423,42 @@ public override void OnRequestPermissionsResult(int requestCode, string[] permis
                 }
             }
         }
+
+        private void turnOffVoice()
+        {
+            voicePrefs.DeleteAllPreferences();
+            voicePrefs.AddString("off");
+            voiceIsOff = true;
+        }
+
+        private void turnOnVoice()
+        {
+            voicePrefs.DeleteAllPreferences();
+            voicePrefs.AddString("on");
+            voiceIsOff = false;
```

Comment:
Note: don't call TtsSpeakAfterInit on Main, because it tells the user the wrong message. You should call the speak that's in OnRestart(), but don't call on OnRestart() itself.